src/DefaultHeadersRequest.jl

```diff
@@ -0,0 +1,62 @@
+module DefaultHeadersRequest
```

Comment:
This is a new "layer" that splits some of the logic out from MessageRequest.jl layer to a separate layer since the MessageRequest.jl layer is now purely for constructing the `Request` object at the start of the stack